src/components/App/App.js

```diff
@@ -0,0 +1,61 @@
+import AllMovies from '../AllMovies/AllMovies';
+import MovieDetails from '../MovieDetails/MovieDetails';
+import Navbar from '../Navbar/Navbar';
+import './App.css';
+import { useState, useEffect } from 'react';
+import "@fontsource/monoton";
+import { getData } from '../.././apiCalls/apiCalls'
+import { checkServerError } from '../../helperFunctions';
+import { Route, Routes } from 'react-router-dom'
+
+const App = () => {
+  const [searchValue, setSearchValue] = useState('');
+  const [movies, setMovies] = useState([]);
+  const [allMovies, setAllMovies] = useState([])
+  const [serverError, setServerError] = useState(false);
+  const [waitingForFetch, setWaitingForFetch] = useState(true);
+
+  let chosenMovieId
+
+  useEffect(() => {
+    getData('movies')
+      .then(data => {
+        if (data.movies) {
+          setMovies(data.movies);
+          setAllMovies(data.movies);
+          setWaitingForFetch(false);
+        } else {
+          setServerError(true);
+          setWaitingForFetch(false);
+        }
+      })
+      .catch((err) => {
+        console.error(err);
+        setServerError(true);
+        setWaitingForFetch(false);
+      })
+  }, [])
+
+  const filterMovies = (keyword, movieList) => {
+    return [...movieList].filter(movie => movie.title.toLowerCase().includes(keyword.toLowerCase()))
+  }
+
+  const changeSearch = value => {
+    setSearchValue(value);
+    setMovies(filterMovies(value, allMovies));
+  }
```

Comment:
For App file level, i would try to keep this clean from component logic. I would move the logic into the components. 

App renders the components, but the AllMovies handles the movies and MovieDetails handles the details. 

If AllMovies and MovieDetails need to share state, another approach is to put this logic into a [custom hook](https://legacy.reactjs.org/docs/hooks-custom.html) that both components can consume.

Reply:
agreed. thanks!!!